### PR TITLE
619: Overriding layouts and templates allows to select an incompatibl…

### PR DESCRIPTION
**Pull request by Iamwade**
**Description** (*)

**Fixed Issues (if relevant)**
Added logic that in the theme selector shows only themes in which this file can be overridden

1. Fixes magento/magento2-phpstorm-plugin#604

**Contribution checklist** (*)
 - [ ] Pull request has a meaningful description of its purpose
 - [ ] All commits are accompanied by meaningful commit messages
 - [ ] All new or changed code is covered with integration/functional tests (if applicable)
 - [ ] All automated tests passed successfully (all builds are green)

### CODE REVIEW

File: src/com/magento/idea/magento2plugin/actions/generation/dialog/OverrideInThemeDialog.java [hunk 1]
```diff
@@ -131,9 +131,13 @@ public static void open(final @NotNull Project project, final PsiFile psiFile) {
     }
 
     private void fillThemeOptions() {
+        final String area = psiFile.getVirtualFile().getPath().split("view/")[1].split("/")[0];
         final List<String> themeNames = new ModuleIndex(project).getEditableThemeNames();
-        for (final String themeName: themeNames) {
-            theme.addItem(themeName);
+        final String baseArea = "base";
+        for (final String themeName : themeNames) {
+            if (baseArea.equals(area) || themeName.split("/")[0].equals(area)) {
```
Comment (bohdan-harniuk) [inline src/com/magento/idea/magento2plugin/actions/generation/dialog/OverrideInThemeDialog.java:138]: Please, use com.magento.idea.magento2plugin.magento.packages.Areas.base.toString() for base area name

File: src/com/magento/idea/magento2plugin/actions/generation/dialog/OverrideInThemeDialog.java [hunk 2]
```diff
@@ -131,9 +131,13 @@ public static void open(final @NotNull Project project, final PsiFile psiFile) {
     }
 
     private void fillThemeOptions() {
+        final String area = psiFile.getVirtualFile().getPath().split("view/")[1].split("/")[0];
         final List<String> themeNames = new ModuleIndex(project).getEditableThemeNames();
-        for (final String themeName: themeNames) {
-            theme.addItem(themeName);
+        final String baseArea = "base";
```
Comment (bohdan-harniuk) [inline src/com/magento/idea/magento2plugin/actions/generation/dialog/OverrideInThemeDialog.java:136]: Please, add an empty line after this one.

**Review by bohdan-harniuk — CHANGES_REQUESTED** [review]
Hello, @Iamwade!

Please, take a look at my comments.

File: src/com/magento/idea/magento2plugin/actions/generation/dialog/OverrideInThemeDialog.java (hunk 2 above)
Comment (Iamwade) [reply src/com/magento/idea/magento2plugin/actions/generation/dialog/OverrideInThemeDialog.java]: Done

File: src/com/magento/idea/magento2plugin/actions/generation/dialog/OverrideInThemeDialog.java (hunk 1 above)
Comment (Iamwade) [reply src/com/magento/idea/magento2plugin/actions/generation/dialog/OverrideInThemeDialog.java]: Fixed

**Comment by Iamwade** [comment]
Hello @bohdan-harniuk 
all points fixed. Thank you!

**Review by bohdan-harniuk — APPROVED** [review]
Hello, @Iamwade!

Thank you for your contribution.

**Pull request merged**